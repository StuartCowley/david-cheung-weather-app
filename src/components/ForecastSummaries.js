import React from "react";
import PropTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";
import "../styles/forecast-summaries.css";

function ForecastSummaries({ forecasts, onForecastSelect }) {
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          temperature={forecast.temperature}
          description={forecast.description}
          icon={forecast.icon}
          onSelect={onForecastSelect}
        />
      ))}
    </div>
  );
}

export default ForecastSummaries;

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.number,
      temperature: PropTypes.shape({
        min: PropTypes.number,
        max: PropTypes.number,
      }),
    })
  ).isRequired,
  onForecastSelect: PropTypes.func.isRequired,
};
