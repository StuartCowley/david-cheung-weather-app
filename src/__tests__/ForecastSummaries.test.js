import React from "react";
import { render, screen } from "@testing-library/react";
import ForecastSummaries from "../components/ForecastSummaries";
import renderer from "react-test-renderer";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1525046400000,
        temperature: {
          max: 11,
          min: 4,
        },
        description: "Clear",
        icon: 800,
      },
      {
        date: 1525132800000,
        temperature: {
          max: 13,
          min: 8,
        },
        description: "Stormy",
        icon: 211,
      },
      {
        date: 1525219200000,
        temperature: {
          max: 1,
          min: -2,
        },
        description: "Heavy Snow",
        icon: 602,
      },
      {
        date: 1525305600000,
        temperature: {
          max: 20,
          min: 4,
        },
        description: "Tornado",
        icon: 781,
      },
      {
        date: 1525392000000,
        temperature: {
          max: 25,
          min: 18,
        },
        description: "Hazy",
        icon: 721,
      },
    ],
    onForecastSelect: () => {},
  };

  test("Renders as expected", () => {
    const rendered = renderer.create(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );

    expect(rendered).toMatchSnapshot();
  });

  test("Assert 5 foreast are present", () => {
    render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    const items = screen.getAllByTestId("forecast-summary");
    expect(items).toHaveLength(5);
  });

  test("Assert 5 button are present", () => {
    render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    const buttons = screen.getAllByRole("button");
    expect(buttons).toHaveLength(5);
  });
});
