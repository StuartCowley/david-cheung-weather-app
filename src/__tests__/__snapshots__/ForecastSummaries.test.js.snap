// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries Renders as expected 1`] = `
<div
  className="forecast-summaries"
>
  <div
    className="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      className="forecast-summary__date"
    >
      Mon 30th Apr
    </div>
    <div
      className="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        className="wi wi-day-sunny"
        name="owm"
      />
    </div>
    <div
      className="forecast-summary__description"
    >
      Clear
    </div>
    <div
      className="forecast-summary__temperature"
    >
      11
      °C
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      More details
    </button>
  </div>
  <div
    className="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      className="forecast-summary__date"
    >
      Tue 1st May
    </div>
    <div
      className="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        className="wi wi-day-thunderstorm"
        name="owm"
      />
    </div>
    <div
      className="forecast-summary__description"
    >
      Stormy
    </div>
    <div
      className="forecast-summary__temperature"
    >
      13
      °C
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      More details
    </button>
  </div>
  <div
    className="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      className="forecast-summary__date"
    >
      Wed 2nd May
    </div>
    <div
      className="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        className="wi wi-day-snow"
        name="owm"
      />
    </div>
    <div
      className="forecast-summary__description"
    >
      Heavy Snow
    </div>
    <div
      className="forecast-summary__temperature"
    >
      1
      °C
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      More details
    </button>
  </div>
  <div
    className="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      className="forecast-summary__date"
    >
      Thu 3rd May
    </div>
    <div
      className="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        className="wi wi-tornado"
        name="owm"
      />
    </div>
    <div
      className="forecast-summary__description"
    >
      Tornado
    </div>
    <div
      className="forecast-summary__temperature"
    >
      20
      °C
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      More details
    </button>
  </div>
  <div
    className="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      className="forecast-summary__date"
    >
      Fri 4th May
    </div>
    <div
      className="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        className="wi wi-day-haze"
        name="owm"
      />
    </div>
    <div
      className="forecast-summary__description"
    >
      Hazy
    </div>
    <div
      className="forecast-summary__temperature"
    >
      25
      °C
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      More details
    </button>
  </div>
</div>
`;
